packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -426,24 +433,90 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         }
     }
 
+    private tooltip(x: number, y: number): void {
+        this.isComputing = true;
+        const startPixel = this.positionXMove;
+        const points: any = [];
+        let zeros = 0;
+        this.state.xyData.datasets.forEach((d: any) => {
+            if (this.state.xyData.datasets.length === 1 || d.data[startPixel] > 0) {
+                const rounded = isNaN(d.data[startPixel]) ? 0 : (BIMath.round(Number(d.data[startPixel]) * 100)/BIMath.round(100));
+                const point: any = {
+                    label: d.label,
+                    color: d.borderColor,
+                    background: d.backgroundColor,
+                    value: rounded.toString(),
+                };
+                points.push(point);
+            }
+            else {
+                zeros += 1;
+            }
+        });
+
+        const tooltipData = {
+            title: this.state.xyData.labels[startPixel],
+            dataPoints: points,
+            top: window.pageYOffset + y,
+            left: window.pageXOffset + x - 200,
```

Comment:
The tooltip position is part of the data, and the tooltip is delayed by 500ms. So what happens is that the user's mouse movements are reproduced after a delay. It makes a very odd behavior. For example, draw a figure 8 with the mouse, the tooltip will also move in a figure 8 but after the delay, even if the mouse has settled on a new position.

I thought of a possible solution, to have two modes:

A) In this mode, the tooltip is updated and shown immediately as the user moves the mouse. Every time the mouse moves, a 500ms debounce is called. When the debounce expires (mouse has stopped moving), we go to mode B.

B) In this mode, we have not moved the mouse for 500 ms and are showing the tooltip. When the user moves the mouse, the tooltip remains where it is and does not update, but a 500ms debounce is called. When the debounce expires, or if the user mouses over the tooltip, we go back to mode A and start updating the tooltip again.

In other words, if the users stops moving for 500ms, then the tooltip will initially not move for 500ms when the user resumes moving.

Reply:
Good suggestions. In B, either using a timer or a debouncer, there would always be a case where the tooltip would behave badly, like staying in one point for longer than needed, and then jumping to another far away point before returning to A.
The solution was to stop the tooltip from following the mouse. Now it shows up after the mouse stops, like the tooltip of the resources status chart.